dataactcore/interfaces/function_bag.py

```diff
@@ -350,4 +350,23 @@ def getErrorMetricsByJobId(job_id, include_file_types=False, severity_id=None):
             record_dict['source_file'] = FILE_TYPE_DICT_ID.get(result.file_type_id, '')
             record_dict['target_file'] = FILE_TYPE_DICT_ID.get(result.target_file_type_id, '')
         result_list.append(record_dict)
-    return result_list
\ No newline at end of file
+    return result_list
+
+def get_cross_file_combos():
+    """Return a list of source and target file types used in cross-file validations."""
+    sess = GlobalDB.db().session
+
+    # note: we query the file_type table here instead of
+    # using FILE_TYPE_DICT in lookups.py because cross-file
+    # validations don't apply to all file types (file types
+    # with a NULL file_order--like E and F--are excluded
+    # from the results of the query below)
+    target_files = sess.query(FileType).subquery()
+    cross_file_combos = sess.query(
+        FileType.name.label('first_file_name'),
+        FileType.file_type_id.label('first_file_type_id'),
+        target_files.c.name.label('second_file_name'),
+        target_files.c.file_type_id.label('second_file_type_id')
+    ).filter(FileType.file_order < target_files.c.file_order)
```

Comment:
Messing around with this a bit...

```python
from itertools import combinations

from dataactcore.models.lookups import FILE_TYPE_DICT

cross_file_exclude_list = ['sub_award', 'awardee_attributes']
cross_file_dict = {file_name: file_id for file_name, file_id in FILE_TYPE_DICT.items() if file_name not in cross_file_exclude_list}

test = list(map(dict, combinations(cross_file_dict.items(), 2)))
```

creates something to work with:
```
[{'award': 4, 'award_financial': 3},
 {'award': 4, 'program_activity': 2},
 {'appropriations': 1, 'award': 4},
 {'award': 4, 'award_procurement': 5},
 {'award_financial': 3, 'program_activity': 2},
 {'appropriations': 1, 'award_financial': 3},
 {'award_financial': 3, 'award_procurement': 5},
 {'appropriations': 1, 'program_activity': 2},
 {'award_procurement': 5, 'program_activity': 2},
 {'appropriations': 1, 'award_procurement': 5}]
```

Overkill?